### PR TITLE
Fixing dtype issues in network.get

### LINKED ISSUE #198: circ_extr.nodes.property_dtypes throws BluepySnapError: Same property with different dtype. model_template: category!= category
Hi,
I got the Zenodo SSCx anatomy Circuit object to create, 
See: https://github.com/BlueBrain/snap/issues/197

but now I'm trying to run the first query command in the example notebook, and 
'circ_extr.nodes.property_dtypes' throws a BluepySnapError: Same property with different dtype. model_template: category!= category
'circ_extr.nodes.property_names'  seems to work fine.

### LINKED ISSUE #203: circ.edges.pair_edges(pre_node, post_node, properties) leads to Error
I run the following:
image

which leads to a BluepySnapError with the following description:

```
File ~/venvs/bpsnap/lib/python3.10/site-packages/bluepysnap/network.py:57, in NetworkObject.property_dtypes.<locals>._update(d, index, value)
     55 def _update(d, index, value):
     56     if d.setdefault(index, value) != value:
---> 57         raise BluepySnapError(
     58             f"Same property with different dtype. {index}: {value}!= {d[index]}"
     59         )

BluepySnapError: Same property with different dtype. afferent_section_id: int16!= int64
```

Source seems to be section ids being represented by different data types in different EdgePopulations, but I think Snap should handle that gracefully.

### PR DESCRIPTION
**Pull request by joni-herttuainen**
Fixes #198 .
Fixes #203 .

Furthermore the following can be closed:  
* #199
* #180 
* #186

### CODE REVIEW

File: bluepysnap/network.py [hunk 1]
```diff
@@ -159,14 +159,7 @@ def get(self, group=None, properties=None):
         if unknown_props:
             raise BluepySnapError(f"Unknown properties required: {unknown_props}")
 
-        # Retrieve the dtypes of the selected properties.
-        # However, the int dtype may not be preserved if some values are NaN.
-        dtypes = {
-            column: dtype
-            for column, dtype in self.property_dtypes.items()
```
Comment (joni-herttuainen) [inline bluepysnap/network.py:166]: Should we remove the whole function definition for `property_dtypes` on the `network` level?

File: bluepysnap/network.py [hunk 2]
```diff
@@ -177,9 +170,9 @@ def get(self, group=None, properties=None):
                 # However, it's a bit more performant than converting the Series to numpy arrays.
                 pop_df = pd.DataFrame({prop: pop.get(pop_ids, prop) for prop in pop_properties})
                 pop_df.index = global_pop_ids.index
-                dataframes.append(pop_df)
-        res = pd.concat(dataframes)
-        assert res.index.is_monotonic_increasing, "The index should be already sorted"
+
+                # Sort the columns in the given order
+                res.append((name, pop_df[[p for p in properties if p in pop_properties]]))
         return res
```
Comment (joni-herttuainen) [inline bluepysnap/network.py:176]: Also, should we return a list, tuple or a generator?

**Review by joni-herttuainen — COMMENTED** [review]
@GianlucaFicarelli : Can you take a quick look and provide your opinion to the questions I raised.

No need to do a full review or spend too much time on this. Just initial thoughts is fine.

Also: I couldn't come up with new scenarios we should be testing for. If anything comes to mind, let me know.

File: bluepysnap/network.py (hunk 1 above)
Comment (GianlucaFicarelli) [reply bluepysnap/network.py:166]: I think so, I don't find it useful anymore

File: bluepysnap/network.py (hunk 2 above)
Comment (GianlucaFicarelli) [reply bluepysnap/network.py:176]: I prefer a generator, so the memory usage can be kept lower when not all the dataframes are needed at the same time.

File: tests/test_edges/test_edges.py [hunk 3]
```diff
@@ -339,8 +343,8 @@ def test_get(self):
         )
         expected = pd.DataFrame(
             {
-                "other2": np.array([np.NaN, np.NaN, np.NaN, np.NaN], dtype=float),
-                "other1": np.array([np.NaN, np.NaN, np.NaN, np.NaN], dtype=object),
+                # "other2": np.array([np.NaN, np.NaN, np.NaN, np.NaN], dtype=float),
```
[the remark this answers is not in the document]
Comment (GianlucaFicarelli) [reply tests/test_edges/test_edges.py:346]: +1

File: CHANGELOG.rst [hunk 4]
```diff
@@ -1,6 +1,19 @@
 Changelog
 =========
 
+Version v1.2.0
+--------------
+
+Breaking Changes
+~~~~~~~~~~~~~~~~
+- ``nodes.get`` and ``edges.get`` (and ``network.get``) no longer return a dataframe
+  - returns a generator yielding tuples of ``(<population_name>, <dataframe>)`` instead
```
Comment (mgeplf) [inline CHANGELOG.rst:10]: maybe add a comment about how one can take that generator, and get the previous behavior?

Comment (joni-herttuainen) [reply CHANGELOG.rst:10]: Added this in changelog and also added some examples in  notebooks (previously not covered) that gives a hint how to  work with `nodes.get` and `edges.get`.

Comment (mgeplf) [reply CHANGELOG.rst:10]: great, thanks

**Review by mgeplf — APPROVED** [review]

**Pull request merged**